runtime/testenv/testenv.go

```diff
@@ -52,11 +53,20 @@ var (
 	env *envtest.Environment
 )
 
-func init() {
-	klog.InitFlags(nil)
+func initLogger() {
+	// NOTE: Set the klog flags on a separate FlagSet to avoid conflicting with
+	// any flags set by the application code. For example, an application
+	// importing glog would conflict with the flags set by klog, resulting in
+	// redeclaration of the same flags.
+	// This is based on klog's coexist with glog example, refer:
+	// https://github.com/kubernetes/klog/blob/f8e668dbaa5f6f0e6a5c24ffd7667263840d79ae/examples/coexist_glog/coexist_glog.go
+	// To override this logger setup, a custom logger can be created and
+	// assigned to the controller-runtime logger explicitly before creating
+	// testenv Environment.
+	klogFlags := flag.NewFlagSet("klog", flag.ExitOnError)
+	klog.InitFlags(klogFlags)
```

Comment:
Do you need to call `klogFlags.Parse()`? I don't think klog calls it.

Reply:
Searching for `Parse()` in klog repo, I only see them in the example code. So, yes, you're probably right about that.

Reply:
I tried setting a parse here `klogFlags.Parse(os.Args[1:])` , but it results in the following error :
```console
flag provided but not defined: -test.paniconexit0
Usage of klog:
  -add_dir_header
        If true, adds the file directory to the header of the log messages
  -alsologtostderr
        log to standard error as well as files
...
```
Because some extra flags get set in the test run that klog flagset isn't aware of.
I can't think of a way to remove unknown flag args before passing to this Parse() in a library. Maybe this is the reason it's not parsed here in the first place.
This leaves it up to the user to construct their own flagset and configure their logger. The library only does the minimum default setup.

Reply:
So that makes me feel like dropping `klog.InitFlags()` completely since the flags are not parsed and used at all.

Reply:
And if we do that, we can remove all the extra comments about coexisting with glog. It'll just work with glog without doing anything.

Reply:
I thought that might be the case. And if it can't be done here, I don't think a user of the library has much of a chance either. It worked before because the test environment didn't mandate using `klogr`; could this package go back to that arrangement?

Reply:
I think it did mandate klogr before but didn't care to parse the flags and left no way to change it because of running the logger setup in the init.
I can just remove the separate flagset code and `klog.InitFlags()`.  That will be equivalent to the previous behavior.